### PR TITLE
[dep] Updgrades dependencies for lmi-dist engine

### PR DESCRIPTION
**Pull request by maaquib**
## Description ##

~~Depends on https://github.com/deepjavalibrary/djl-serving/pull/1934~~
Adding `max_model_len` param to ci tests [vllm-project/vllm/issues/2418]

* Nightly build: https://github.com/deepjavalibrary/djl-serving/actions/runs/9215590750
* Large model integration: https://github.com/deepjavalibrary/djl-serving/actions/runs/9214783969
* P4D integ tests: https://github.com/deepjavalibrary/djl-serving/actions/runs/9214819854

### CODE REVIEW

File: serving/docker/dockerd-entrypoint-with-cuda-compat.sh [hunk 1]
```diff
@@ -19,7 +19,7 @@ translateTGIToLMI() {
 if [ -f /usr/local/cuda/compat/libcuda.so.1 ]; then
     CUDA_COMPAT_MAX_DRIVER_VERSION=$(readlink /usr/local/cuda/compat/libcuda.so.1 |cut -d'.' -f 3-)
     echo "CUDA compat package requires Nvidia driver ⩽${CUDA_COMPAT_MAX_DRIVER_VERSION}"
-    NVIDIA_DRIVER_VERSION=$(sed -n 's/^NVRM.*Kernel Module *\([0-9.]*\).*$/\1/p' /proc/driver/nvidia/version 2>/dev/null || true)
+    NVIDIA_DRIVER_VERSION=$(nvidia-smi --query-gpu=driver_version --format=csv,noheader --id=0 2>/dev/null | cut -d'.' -f1 || true)
```
Comment (lanking520) [inline serving/docker/dockerd-entrypoint-with-cuda-compat.sh:22]: what is this for?

Comment (maaquib) [reply serving/docker/dockerd-entrypoint-with-cuda-compat.sh:22]: I tried the sed command, but it doesn't work, returns an empty string

Comment (rohithkrn) [reply serving/docker/dockerd-entrypoint-with-cuda-compat.sh]: existing command didn't work for me, probably they changed the version file?

If we want to use the version file, this worked `grep -oP '(\d+\.\d+\.\d+)' /proc/driver/nvidia/version | sed -n '1p'`.
I prefer using `nvidia-smi` as that's more standard.

File: serving/docker/lmi.Dockerfile [hunk 2]
```diff
@@ -9,31 +9,33 @@
 # or in the "LICENSE.txt" file accompanying this file. This file is distributed on an "AS IS"
 # BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied. See the License for
 # the specific language governing permissions and limitations under the License.
-ARG version=12.1.1-cudnn8-devel-ubuntu22.04
+ARG version=12.4.1-devel-ubuntu22.04
```
Comment (frankfliu) [inline serving/docker/lmi.Dockerfile:12]: Why remove cudnn8?

Comment (frankfliu) [reply serving/docker/lmi.Dockerfile:12]: should we manually install cudnn?

Comment (maaquib) [reply serving/docker/lmi.Dockerfile:12]: Going with the same base as vllm: https://github.com/vllm-project/vllm/blob/v0.4.2/Dockerfile#L10

Comment (xyang16) [reply serving/docker/lmi.Dockerfile:12]: Onnxruntime need cudnn

Comment (maaquib) [reply serving/docker/lmi.Dockerfile:12]: Will add it back

**Review by lanking520 — APPROVED** [review]

**Review by rohithkrn — APPROVED** [review]

**Pull request merged**